python/Passes.cpp

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mlir/InitAllTranslations.h"
+#include "ttmlir/Bindings/Python/TTMLIRModule.h"
+#include "ttmlir/RegisterAll.h"
+#include "ttmlir/Target/TTNN/TTNNToFlatbuffer.h"
+
+namespace mlir::ttmlir::python {
+
+void populatePassesModule(py::module &m) {
+  // When populating passes, need to first register them
+
+  mlir::tt::registerAllPasses();
+  mlir::registerAllTranslations();
+  // mlir::tt::ttnn::registerTTNNPipelines();
+
+  m.def("ttir_to_ttnn_backend_pipeline", [](MlirModule module) {
+    mlir::Operation *moduleOp = unwrap(mlirModuleGetOperation(module));
+    mlir::PassManager pm(moduleOp->getName());
+
+    mlir::DialectRegistry registry;
+    mlir::tt::registerAllDialects(registry);
+    mlir::MLIRContext *ctx = unwrap(mlirModuleGetContext(module));
+    ctx->appendDialectRegistry(registry);
+
+    const auto pipeline =
+        mlir::PassPipelineInfo::lookup("ttir-to-ttnn-backend-pipeline");
+
+    std::string options = "";
+
+    mlir::function_ref<mlir::LogicalResult(const llvm::Twine &)> err_handler =
+        [](const llvm::Twine &loc) { return mlir::failure(); };
+
+    if (mlir::failed(pipeline->addToPipeline(pm, options, err_handler))) {
+      throw std::runtime_error("Failed to add pipeline to pass manager");
+    }
+
+    if (mlir::failed(pm.run(moduleOp))) {
+      throw std::runtime_error("Failed to run pass manager");
+    }
+  });
+
+  py::class_<std::shared_ptr<void>>(m, "TTNNToFlatbufferData")
+      .def(py::init<>())
+      .def("from_module", [](MlirModule module) {
+        mlir::Operation *moduleOp = unwrap(mlirModuleGetOperation(module));
+        auto data = mlir::tt::ttnn::ttnnToFlatbuffer(moduleOp);
+      });
+
+  m.def("ttnn_to_flatbuffer", [](MlirModule module) {
+    mlir::Operation *moduleOp = unwrap(mlirModuleGetOperation(module));
+    std::shared_ptr<void> data = mlir::tt::ttnn::ttnnToFlatbuffer(moduleOp);
+    return py::capsule(data.get());
```

Comment:
I don't think this will work, can we just return `data`? It _might_ convert automatically to a `TTNNToFlatbufferData`?  If this doesn't work, sync with me and we can create a wrapper type.

The reason this doesn't work is that once the shared ptr `data` leaves this scope, it'll free the memory and the capsule will point to freed memory.

Reply:
Ah right, so the data doesn't automatically transform into the `TTNNToFlatbufferData`. I didn't want to go wrap the pointer "again" since it'll be wrapped into `Binary` from tt::runtime when consumed upstream anyways. 

I was confused about how to return the variable haha, unfortunately `py::capsule` doesn't work nicely with smart pointers. `TTNNToFlatbufferData` was supposed to be removed, so I'll add that change in now. Makes sense that it would point to freed data, I'll take a look at what I can do and we can totally sync :)